presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/common/Kerberos.java

```diff
@@ -36,22 +36,26 @@
 
     private final String hadoopBaseImage;
     private final String hadoopImagesVersion;
+    private final Hadoop hadoop;
 
     @Inject
     public Kerberos(
             DockerFiles dockerFiles,
             PortBinder portBinder,
+            Hadoop hadoop,
```

Comment:
Kerberos -> KerberizedHadoop

?

Reply:
The problem IMO is that Kerberos is not some kind of addition to Hadoop. It is a standalone being.
We have 3 entities here:

- Hadoop
- Kerberos
- Hadoop-Kerberos relationship

Kerberos can be used with other systems. If we add ranger to that we have: Hadoop, Kerberos, ranger and two relationships - ranger-kerberos and hadoop-kerberos.
Expressing this in simple object composition is not trivial.

Reply:
In general yes, but in particular no.

Standalone Hadoop would use standalone thing like `https://github.com/prestosql/docker-images/tree/master/prestodev/kerberos` which is not the case here. Here we are modifying `Hadoop` to make it kerberized, so name `KerberizedHadoop` is great here as there is no standalone `Kerberos` here yet. This even more stresses the fact that injecting `Hadoop` and `Kerberos` previously was bad.

I see your point with entities here, and I agree with you. Best option if we would have containers `A` and `B` and the `C` that affects `A` and `B` to be expressed a collection  of `AwithC` (that depends on `A` and `C`), `BwithC` (that depends on `B` and `C`). Do we agree here?

Reply:
`C` was representing kerberos here. But please notice that current `Kerberos` does not represent standalone component, it is just a modification.

Reply:
Let's start from the beginning. This is depicting the dependency between these 2 as they are right now. We could model them in a more granular manner, but right not we should start from making dependencies like this visible and naming the abstractions properly (like `KerberizedHadoop`), so that they start to represent what they actually do.